### PR TITLE
Migrate simple monitor commands to MonitorManager

### PR DESCRIPTION
**Pull request by t-wissmann**
This moves the commands

    focus_monitor, cycle_monitor, shift_to_monitor, monitor_rect

to MonitorManager and makes use of ArgParse. The existing test cases for
these commands are extended such that parse errors and tab completion
are also covered.

For these commands, the type-safety and automation of ArgParse comes in
quite handy.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ unique_ptr<CommandTable> commands(shared_ptr<Root> root) {
         {"remove_monitor", { monitors, &MonitorManager::removeMonitorCommand }},
         {"move_monitor",   monitors->byFirstArg(&Monitor::move_cmd, &Monitor::move_complete) } ,
         {"rename_monitor", monitors->byFirstArg(&Monitor::renameCommand, &Monitor::renameComplete) },
-        {"monitor_rect",   monitor_rect_command},
```
Comment (blarz) [inline src/main.cpp:169]: Remove monitor_rect_command in .cpp file in line 367?

**Review by blarz — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (blarz) [reply src/main.cpp:169]: Or is it still needed?

Comment (t-wissmann) [reply src/main.cpp:168]: Thanks! I simply forgot to remove it. This was probably the reason why the coverage decreased by 0.1%.

Comment (blarz) [reply src/main.cpp:168]: Makes sense :D Looks good now, thanks!

**Review by blarz — APPROVED** [review]
Very nice!

**Pull request merged**